nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -307,6 +313,41 @@ private UpdateableReference loadReference(String requestedRef) {
     }
   }
 
+  private UpdateableReference loadReferenceAtTime(String requestedRef, Instant timestamp) {
+    try {
+      Reference ref = requestedRef == null ? client.getTreeApi().getDefaultBranch()
+          : client.getTreeApi().getReferenceByName(requestedRef);
+      if (ref instanceof Hash) {
+        LOGGER.warn("Cannot specify a hash {} and timestamp {} together. " +
+            "The timestamp is redundant and has been ignored", requestedRef, timestamp);
+        return new UpdateableReference(ref, client.getTreeApi());
```

Comment:
We might want to change the name of `UpdateableReference` because it is misleading in a context like this. The reference here isn't actually updateable because it is a hash. I think the class is named because it _can_ update _some_ references.